[WIP] IBM MQ source

sebgoa opened the pull request:
This is a proposed new container source, it gets events from IBM MQ queues.

It is self-contained and moved as is in `/contrib`

It does not use `ko` yet.

mattmoor-sockpuppet commented on contrib/ibm-mq/Makefile:
```diff
@@ -0,0 +1,15 @@
+	golint ./...
+
+vet:
+	go vet ./...
```
<!--whitespace-bot-->
```suggestion
	go vet ./...

```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
	go vet ./...

```

mattmoor-sockpuppet commented on contrib/ibm-mq/README.md:
````diff
@@ -0,0 +1,36 @@
+
+```
+kubectl apply -f mq-source.yaml
+```
````
<!--whitespace-bot-->
```suggestion
```

```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
```

```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
```

```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
```

```

mattmoor-sockpuppet commented on contrib/ibm-mq/mq-source.yaml:
```diff
@@ -0,0 +1,24 @@
+      value: admin
+    - name: PASSWORD
+      value: passw0rd
+    - name: QUEUE_NAME 
```
<!--whitespace-bot-->
```suggestion
    - name: QUEUE_NAME
```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
    - name: QUEUE_NAME
```

mattmoor-sockpuppet replied:
<!--whitespace-bot-->
```suggestion
    - name: QUEUE_NAME
```

evankanderson commented:
I suggest you add an OWNERS here pointing to yourself, so that you and co-maintainers can review subsequent updates without needing upstream reviewers.

See https://github.com/knative/eventing-sources/blob/master/contrib/kafka/OWNERS and https://github.com/knative/eventing-sources/blob/master/OWNERS_ALIASES for an example

sebgoa commented:
@evankanderson thanks for looking. I added a OWNERS file and modified the OWNER_ALIASES file.
Fixed the copyright.

Because this needs some bindings, I don't see how we can use `ko` for this, so we have to add a Dockerfile.

I will need to figure out how we can tie this to automatically pushing a container. Right now we are referencing a container in `gcr.io/triggermesh` project.

I know @duglin is having a look at this as well.

n3wscott commented:
/retest

vaikas-google commented:
Looks like it's maybe missing some dependencies?

I0604 20:17:22.470] Running tests with 'go test -race -v  ./... '
I0604 20:17:24.992] contrib/ibm-mq/main.go:25:2: cannot find package "github.com/caarlos0/env" in any of:
I0604 20:17:24.995] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/caarlos0/env (vendor tree)
I0604 20:17:24.995] 	/usr/local/go/src/github.com/caarlos0/env (from $GOROOT)
I0604 20:17:24.995] 	/go/src/github.com/caarlos0/env (from $GOPATH)
I0604 20:17:24.998] contrib/ibm-mq/main.go:26:2: cannot find package "github.com/ibm-messaging/mq-golang/ibmmq" in any of:
I0604 20:17:25.001] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/ibm-messaging/mq-golang/ibmmq (vendor tree)
I0604 20:17:25.002] 	/usr/local/go/src/github.com/ibm-messaging/mq-golang/ibmmq (from $GOROOT)
I0604 20:17:25.002] 	/go/src/github.com/ibm-messaging/mq-golang/ibmmq (from $GOPATH)
I0604 20:17:25.003] contrib/ibm-mq/main.go:27:2: cannot find package "github.com/knative/pkg/cloudevents" in any of:
I0604 20:17:25.003] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/knative/pkg/cloudevents (vendor tree)
I0604 20:17:25.004] 	/usr/local/go/src/github.com/knative/pkg/cloudevents (from $GOROOT)
I0604 20:17:25.005] 	/go/src/github.com/knative/pkg/cloudevents (from $GOPATH)
I0604 20:17:25.021] Finished run, return code is 1
I0604 20:17:25.063] XML report written to /workspace/_artifacts/junit_IIOBcBiN.xml

sebgoa commented:
This does not build with `ko`, we need to run `dep ensure` to get the dependencies before the `go build`

mattmoor-sockpuppet commented on contrib/ibm-mq/README.md:
```diff
@@ -0,0 +1,105 @@
+
+Using _Admin_ and _passw0rd_ as default development credentials do:
+
+`open https://localhost:9443/ibmmq/console/` 
```
<!--whitespace-bot-->
```suggestion
`open https://localhost:9443/ibmmq/console/`
```

mattmoor-sockpuppet commented on contrib/ibm-mq/README.md:
````diff
@@ -0,0 +1,105 @@
+
+```
+kubectl apply -f mq-source.yaml
+```
````
<!--whitespace-bot-->
```suggestion
```

```

mattmoor-sockpuppet commented on contrib/ibm-mq/README.md:
```diff
@@ -0,0 +1,108 @@
+
+Using _Admin_ and _passw0rd_ as default development credentials do:
+
+`open https://localhost:9443/ibmmq/console/` 
```
<!--whitespace-bot-->
```suggestion
`open https://localhost:9443/ibmmq/console/`
```

mattmoor-sockpuppet commented on contrib/ibm-mq/README.md:
````diff
@@ -0,0 +1,108 @@
+
+```
+kubectl apply -f mq-source.yaml
+```
````
<!--whitespace-bot-->
```suggestion
```

```

sebgoa commented:
closing in favor of #500
